### PR TITLE
[4.3] Convert content fields plugin to services

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Converts the content fields plugin to services.

### Testing Instructions
- Create an article custom field
- Create an article and set as content "Field demo {field 1}".
- Define a value for the custom field in the article form
- Open the article on the front

### Actual result BEFORE applying this Pull Request
The field is rendered within the article description.

### Expected result AFTER applying this Pull Request
The field is rendered within the article description.

### CODE REVIEW

File: plugins/content/fields/services/provider.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        $container->set(
+            PluginInterface::class,
+            function (Container $container) {
+                $dispatcher = $container->get(DispatcherInterface::class);
+                $plugin     = new Fields(
+                    $dispatcher,
+                    (array) PluginHelper::getPlugin('content', 'fields')
+                );
+                $plugin->setApplication(Factory::getApplication());
```
Comment (heelc29) [inline plugins/content/fields/services/provider.php]: @laoneo The _Application object_ is not used by the plugin. Should it still be injected into the plugin?

Comment (laoneo) [reply plugins/content/fields/services/provider.php]: I'm not sure if it is still used indirectly, does all work when you remove it?

Comment (heelc29) [reply plugins/content/fields/services/provider.php]: > does all work when you remove it?

Yes, the output is still the same.
image

Comment (laoneo) [reply plugins/content/fields/services/provider.php]:
```suggestion
```

Comment (laoneo) [reply plugins/content/fields/services/provider.php]: Ok, removed the application injection

File: plugins/content/fields/services/provider.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * @subpackage  Content.fields
+ *
+ * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Extension\PluginInterface;
+use Joomla\CMS\Factory;
```
Comment (heelc29) [inline plugins/content/fields/services/provider.php]:
```suggestion
```
Please also remove the unused import.

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on f598104fdc1a9013ba28e3b3c3b07e44c2dbf58a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38470">issues.joomla.org/tracker/joomla-cms/38470</a>.</sub>

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on a2cab0beb4a509f263b6891de2c1ebf6f701a41b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38470">issues.joomla.org/tracker/joomla-cms/38470</a>.</sub>

**Comment by obuisard** [comment]
Thank you Allon @laoneo for the PR

**Pull request merged**